wrapper/src/main/java/software/amazon/jdbc/plugin/customendpoint/CustomEndpointMonitorImpl.java

```diff
@@ -34,92 +31,85 @@
 
   // Keys are custom endpoint URLs, values are information objects for the associated custom endpoint.
   protected static final CacheMap<String, CustomEndpointInfo> customEndpointInfoCache = new CacheMap<>();
   protected static final long CUSTOM_ENDPOINT_INFO_EXPIRATION_NANO = TimeUnit.MINUTES.toNanos(5);
 
-  protected final AtomicBoolean stop = new AtomicBoolean(false);
   protected final RdsClient rdsClient;
   protected final HostSpec customEndpointHostSpec;
   protected final String endpointIdentifier;
   protected final Region region;
   protected final long refreshRateNano;
-
-  protected final PluginService pluginService;
-  protected final ExecutorService monitorExecutor = Executors.newSingleThreadExecutor(runnableTarget -> {
-    final Thread monitoringThread = new Thread(runnableTarget);
-    monitoringThread.setDaemon(true);
-    if (!StringUtils.isNullOrEmpty(monitoringThread.getName())) {
-      monitoringThread.setName(monitoringThread.getName() + "-cem");
-    }
-    return monitoringThread;
-  });
+  protected final StorageService storageService;
 
   private final TelemetryCounter infoChangedCounter;
 
   /**
    * Constructs a CustomEndpointMonitorImpl instance for the host specified by {@code customEndpointHostSpec}.
    *
-   * @param pluginService          The plugin service to use to update the set of allowed/blocked hosts according to
-   *                               the custom endpoint info.
+   * @param monitorService         The monitorService used to submit this monitor.
+   * @param storageService         The storage service used to store the set of allowed/blocked hosts according to the
+   *                               custom endpoint info.
+   * @param telemetryFactory       The telemetry factory used to create telemetry data.
    * @param customEndpointHostSpec The host information for the custom endpoint to be monitored.
-   * @param endpointIdentifier An endpoint identifier.
+   * @param endpointIdentifier     An endpoint identifier.
    * @param region                 The region of the custom endpoint to be monitored.
    * @param refreshRateNano        Controls how often the custom endpoint information should be fetched and analyzed for
    *                               changes. The value specified should be in nanoseconds.
    * @param rdsClientFunc          The function to call to create the RDS client that will fetch custom endpoint
    *                               information.
    */
   public CustomEndpointMonitorImpl(
-      PluginService pluginService,
+      CoreMonitorService monitorService,
+      StorageService storageService,
+      TelemetryFactory telemetryFactory,
       HostSpec customEndpointHostSpec,
       String endpointIdentifier,
       Region region,
       long refreshRateNano,
       BiFunction<HostSpec, Region, RdsClient> rdsClientFunc) {
-    this.pluginService = pluginService;
+    super(30);
```

Comment:
Probably it's better to use a constant with human readable name rather than cryptic "30". It's a nit anyway.

Reply:
Good catch, this was uninentional